lib/utils.js

```diff
@@ -146,3 +150,56 @@ utils.registerSourceMap = () => {
 utils.registerTypeScript = () => {
   require('ts-node').register();
 };
+
+/**
+ * Check if given template is installed and return details about it
+ * 
+ * @private
+ * @param {String} name name of the template
+ * @param {String} PACKAGE_JSON_FILENAME standard name of the package.json file
+ * @returns {Object|undefined} object representing content of the package.json file with additional field -> pkgPath (absolute path to the package)
```

Comment:
@jonaslagoni @magicmatatjahu if you feel it is important for JSDoc to properly specify package.json object, I found https://github.com/sindresorhus/type-fest/blob/main/source/package-json.d.ts. Maybe there is a way to somehow link both worlds? I can research

Reply:
I think description tell what exactly is inside the object.

Reply:
The problem is you would need to know what properties are part of the `package.json` file. From the maintainer's point of view, I would rather not have to go to some docs to find the correct properties 😄

Reply:
By using those types you would be able to write something like this: 
```
/**
 * @typedef {Object} ResolveType
 * @prop {string} pkgPath absolute path to the package
 */

 * @returns {PackageJson & ResolveType | undefined} object representing content of the package.json file with additional field -> pkgPath (absolute path to the package)
 
```

Reply:
@jonaslagoni is this what you're looking for:

```
/**
 * @typedef {Object} ResolveType
 * @prop {string} pkgPath absolute path to the package
 */

/**
 * Check if given template is installed and return details about it located in the package.json
 * 
 * @private 
 * @param {String} name name of the template
 * @param {String} PACKAGE_JSON_FILENAME standard name of the package.json file
 * @returns {ResolveType & PackageJson | undefined} object representing content of the package.json file with pkgPath property. Read {@link https://raw.githubusercontent.com/sindresorhus/type-fest/main/source/package-json.d.ts|PackageJson} for more detaisl on contents of the package.json object
 */
```

![Screenshot 2021-03-12 at 12 42 53](https://user-images.githubusercontent.com/6995927/110935996-7c2e2580-8330-11eb-8a1f-4101854978e7.png)

`PackageJson` is still `any`. I would have to add type-fest to dependencies or copy-paste the type to the generator I think. Do we really want it?

Reply:
🤷 I recently had to add the parser types to ts-nats-template because I wanted to have the actual types accessible. Meaning I had to add the parser as a dependency. Not optimal but in my case, I found it necessary to improve maintainability. If you want to add it is up to you 😄 There are pro and cons for both sides.

Reply:
well, if it would be up to me, I would just do what I posted above as I believe it is enough 😄 but I'm happy to do more if you feel it is important. I just think your use case with parser is different. In the template, you operate on parser API a lot and I see a lot of value in importing types definition, but it is much different here in the generator with package.json, we need it only for the name and version and that is it

Reply:
Yep totally fair :) Up to you if you want `ResolveType` but if you don't use `PackageJson` I would remove it and add something like `Object` 😄

Reply:
the problem with `PackageJSON` is that it's a TS type, so most likely JSDoc won't treat it as `typedef`

Reply:
@jonaslagoni @magicmatatjahu ok, I made just a small improvement then. Please approve again